forge-gui/res/cardsfolder/upcoming/braids_arisen_nightmare.txt

```diff
@@ -0,0 +1,13 @@
+Name:Braids, Arisen Nightmare
+ManaCost:1 B B
+Types:Legendary Creature Nightmare
+PT:3/3
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigSac | TriggerDescription$ At the beginning of your end step, you may sacrifice an artifact, creature, enchantment, land, or planeswalker. If you do, each opponent may sacrifice a permanent that shares a card type with it. For each opponent who doesn't, that player loses 2 life and you draw a card.
+SVar:TrigSac:DB$ Sacrifice | Optional$ True | SacValid$ Artifact,Creature,Enchantment,Land,Planeswalker | Amount$ 1 | RememberSacrificed$ True | SubAbility$ DBRepeatEach
+SVar:DBRepeatEach:DB$ RepeatEach | RepeatSubAbility$ DBMaySac | RepeatPlayers$ Player.Opponent | SubAbility$ DBCleanup
```

Comment:
this line needs a condition... opponents don't get to sacrifice unless you already have

you might be able to just make this an AB$ line with a cost and remove line 6 - might take a bit of doing though

Reply:
Right, looks like that case wasn't as airtight as I thought! I've fixed this by adding a `ConditionDefined$ Remembered | ConditionPresent$ Card` clause to halt the `RepeatEach` if nothing is sacrificed. I'm sure it could accommodate the AB$ + Cost solution, but this seemed like the path of least resistance and it seems to hold up. Thanks for the catch